pyproject.toml

```diff
@@ -0,0 +1,6 @@
+[build-system]
+requires = ["setuptools >=42", "setuptools_scm[toml] >=3.4", "wheel"]
```

Comment:
```suggestion
requires = ["setuptools >=45", "setuptools_scm[toml] >=6.3.1", "wheel"]
```

Reply:
Any particular reason to choose these versions? E.g. are there new versions in setuptools 45 and setuptools_scm 6.3.1 that we're likely to use?

Reply:
Primarily Bugfixes in both setuptools and setuptools_scm

Reply:
Well... personally I'm a little reluctant to do this just on account of bug fixes. I mean, I'm thinking that most people who install `pytest-localserver` are just going to get the latest versions of `setuptools` and `setuptools_scm` anyway, because that's what pip installs by default. The only people who would be affected by this change are people who, for some reason, are stuck using old versions of these packages i.e. `setuptools >=42, <45` or `setuptools_scm >=3.4, <6.3.1`. And if those old versions of the packages are sufficient to build `pytest-localserver`, why should we prevent people from using them? If there is a strong reason to prevent use of those versions, sure, I'm not opposed to changing them, but otherwise I'm not really convinced.

I think when I came up with these numbers (for another project), I looked into which versions of `setuptools` and `setuptools_scm` would _actually_ be required. E.g. version 42 of `setuptools` was the one that added support for `pyproject.toml` in the first place. I don't remember why I chose version 3.4 of `setuptools_scm`, but maybe it was the same reason - you probably know better than I do what was new in that version that would have prompted me to list it here.

I can take a look through the history of both projects, and maybe once I get better acquainted with them, this change will make more sense to me. But it'll take a bit of time.

Reply:
the modern peps hand build isolation for the source packages and any normal users should use wheels

wheels will absolutely not need the build requirements installed for usage, thus i dont understand how this can possibly be a issue

Reply:
Wheels don't matter. I'm talking about building from source.

Ultimately what I'm looking for, as justification for these version numbers, is that building the package from source fails with any lower version. Basically, if I'm going to disallow setuptools 42 and setuptools_scm 3.4 then I'd want to see one of the following commands fail:

```console
$ git clone git@github.com:pytest-dev/pytest-localserver.git
$ cd pytest-localserver
$ python -m venv venv
$ venv/bin/pip install setuptools==42 'setuptools_scm[toml]==3.4' wheel build # simulate creating isolated build env
$ venv/bin/pyproject-build --no-isolation .
$ tox --installpkg dist/pytest-localserver-0.5.2.dev7+g1cb8237.tar.gz -r
$ tox --installpkg dist/pytest_localserver-0.5.2.dev7+g1cb8237-py3-none-any.whl -r
```

I did test this with a few versions and found that the commands _do_ fail as written above, but bumping setuptools_scm to 3.5 is sufficient for everything to work as far as I can tell. Same with every higher version that I tested (except on Python 2.7 but I don't care about that). So I'm actually inclined to do this:

```suggestion
requires = ["setuptools >=42", "setuptools_scm[toml] >=3.5", "wheel"]
```

Reply:
My latest push sets the minimum required `setuptools_scm` version at 3.4.1. I've verified using `pyproject-build` that the listed versions are sufficient to build an sdist and wheel with `pyproject-build` and to install and run tests.

Should I consider this comment a blocker, or can I proceed with the PR?

Reply:
please proceed